benches/languages.rs

```diff
@@ -207,6 +208,67 @@ fn benchmark_parse_syslog(c: &mut Criterion) {
     benchmark_configs(c, "parse_syslog", configs, "in", "last", input, output);
 }
 
+fn benchmark_multifaceted(c: &mut Criterion) {
+    let configs: Vec<(&str, &str)> = vec![
+        (
+            "remap",
+            indoc! {r#"
+                [transforms.last]
+                  type = "remap"
+                  inputs = ["in"]
+                  source = """
+                  . = parse_syslog!(.message)
+                  .timestamp = format_timestamp!(.timestamp, format: "%c")
+                  """
+            "#},
+        ),
+        (
+            "lua",
+            indoc! {r#"
+                [transforms.last]
+                  type = "lua"
+                  inputs = ["in"]
+                  version = "2"
+                  source = """
+                  local function parse_and_transform(message)
+                    local pattern = "^<(%d+)>(%d+) (%S+) (%S+) (%S+) (%S+) (%S+) (%S+) (.+)$"
+                    local priority, version, timestamp, _host, appname, procid, msgid, _sdata, message = string.match(message, pattern)
+
+                    local year, month, day, hour, minute, second, tz = string.match(timestamp, '(%d%d%d%d)-(%d%d)-(%d%d)T(%d%d):(%d%d):(%d%d).(%d%d%d)(%a)')
+
+                    timestamp = os.date('%c', os.time({year=year, month=month, day=day, hour=hour, minute=minute, second=second}))
+                    severity = "info"
+
+                    return {severity = severity, priority = priority, version = version, timestamp = timestamp, appname = appname, procid = procid, msgid = msgid, message = message}
+                  end
+
+                  function process(event, emit)
+                    event.log = parse_and_transform(event.log.message)
+                    emit(event)
+                  end
```

Comment:
Could these be combined to avoid the function call overhead?

Reply:
On the one hand, yes. On the other hand, I suspect that this might be a more idiomatic/"in the wild" way to use Lua and perhaps better for benchmarking? No strong feelings on that, though.